fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/api/DelinquencyApiResource.java

```diff
@@ -0,0 +1,217 @@
+    @Operation(summary = "Retrieve a specific Delinquency Bucket based on the Id", description = "")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = DelinquencyApiResourceSwagger.GetDelinquencyBucketsResponse.class))) })
+    public String getDelinquencyBucket(
+            @PathParam("delinquencyBucketId") @Parameter(description = "delinquencyBucketId") final Long delinquencyBucketId,
+            @Context final UriInfo uriInfo) {
+        securityContext.authenticatedUser().validateHasReadPermission("DELINQUENCY_BUCKET");
+        final DelinquencyBucketData delinquencyBucketData = this.readPlatformService.retrieveDelinquencyBucket(delinquencyBucketId);
+        ApiRequestJsonSerializationSettings settings = parameterHelper.process(uriInfo.getQueryParameters());
+        return this.jsonSerializerBucket.serialize(settings, delinquencyBucketData);
+    }
+
+    @POST
+    @Path("buckets")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Create Delinquency Bucket", description = "")
+    @RequestBody(required = true, content = @Content(schema = @Schema(implementation = DelinquencyApiResourceSwagger.PostDelinquencyBucketRequest.class)))
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = DelinquencyApiResourceSwagger.PostDelinquencyBucketResponse.class))) })
+    public String createDelinquencyBucket(final String jsonRequestBody, @Context UriInfo uriInfo) {
+        securityContext.authenticatedUser().validateHasCreatePermission("DELINQUENCY_BUCKET");
+        final CommandWrapper commandRequest = new CommandWrapperBuilder().createDelinquencyBucket().withJson(jsonRequestBody).build();
+
+        CommandProcessingResult result = commandWritePlatformService.logCommandSource(commandRequest);
+        return jsonSerializerRange.serialize(result);
+    }
+
+    @PUT
+    @Path("buckets/{delinquencyBucketId}")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Update Delinquency Bucket based on the Id", description = "")
+    @RequestBody(required = true, content = @Content(schema = @Schema(implementation = DelinquencyApiResourceSwagger.PostDelinquencyBucketRequest.class)))
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = DelinquencyApiResourceSwagger.PutDelinquencyBucketResponse.class))) })
+    public String updateDelinquencyBucket(
+            @PathParam("delinquencyBucketId") @Parameter(description = "delinquencyBucketId") final Long delinquencyBucketId,
+            final String jsonRequestBody, @Context UriInfo uriInfo) {
+        securityContext.authenticatedUser().validateHasCreatePermission("DELINQUENCY_BUCKET");
```

Comment:
@josehernandezfintecheandomx Wrong permission got checked. It should use the "validateHasUpdatePermission"

Reply:
Done!